### ISSUE TITLE
Weights

### ISSUE THREAD

**Issue by ayushpatnaikgit** [xKDR/CRRao.jl#94]
How can we incorporate weights in the models? I want to do bootstrap to calculate the standard errors. The weights will be drawn from the multinomial distribution.

**Comment by sourish-cmi** [comment]
Can you please provide more detail about the models you are talking about? Currently, your question bit vague.

**Comment by ayushpatnaikgit** [comment]
For any model. 
Suppose I have a data frame with 3 columns, X, Y and weights. 
```julia
df = DataFrame(X = 1:10, Y = 20:29, weights = 1:10)
```

In GLM.jl, I can do: 
```julia
glm(@formula(Y ~ X), df, Normal(), IdentityLink(), wts=df.weights)
```
However, in CRRao, there is no option for `wts`. We'll need to add keyword arguments in CRRao that will be passed to GLM. Not sure how weights are considered for bayesian regression. 

With bootstrap, what I want to do is: 
```julia
using Distributions
replicates = 1000
dist_coef = [coef(glm(@formula(Y ~ X), df, Normal(), IdentityLink(), wts=rand(Multinomial(10, 10))))[1] for i in 1:replicates]
SE = std(dist_coef)
```

**Comment by sourish-cmi** [comment]
Yes, this feature can be added. GLM can be added quickly. Not sure how to handle the Bayesian model at this stage.

**Comment by ayushpatnaikgit** [comment]
@ShouvikGhosh2048 can you allow fit to accept keyword arguments and then pass them to glm?

**Comment by ShouvikGhosh2048** [comment]
Should we allow arbitrary GLM keyword arguments?

Like this: 
```Julia
function fit(formula::FormulaTerm, data::DataFrame, modelClass::LinearRegression; kwargs...)
    model = lm(formula, data; kwargs...)
    return FrequentistRegression(:LinearRegression, model, formula)
end
```

Or just weights?

**Comment by ayushpatnaikgit** [comment]
I think all kwargs like you have shown.
